src/routes/fastPayment.js

```diff
@@ -18,18 +19,24 @@ route.post('/fastPayment',
         const price = req.body.price;
         const filter = { userID: req.body.userID };
 
-        userDB.findOne(filter, (err, user) => {
-            if (err) {
-                console.log(err);
-                res.status(500);
-                res.send("Error Has Occured");
-            } else {
-                const result = makePayment(price, user);
-                res.json({ user }); //test purposes!
+        const user = await userDB.findOne(filter);
 
+        try {
+            const iyzicoResult = await makePayment(price, user);
+            const paymentInfo = {
+                paymentID: iyzicoResult.paymentId,
+                status: iyzicoResult.status
             }
-
-        });
+            if (iyzicoResult.status === 'success') {
+                const payment = await paymentDB.findOneAndUpdate(filter, paymentInfo, { new: true });
```

Comment:
payment record should be created here. I didn't understand what we are updating.
After makePayment is done, we should create a new DB record for this payment to use it in future.
and also line 31 and 34 are exactly same so no need to have both of them just make it after if-else block.